### PR TITLE
Correct index.lock file deletion and other fixes & formatting

### PR DESCRIPTION
**Pull request by sandipagarwal**
The following have been fixed.
1. [index.lock](https://github.com/sebdah/git-pylint-commit-hook/commit/d82fb72191ecf06e3189c4ce5083d18a7771abad#commitcomment-19130579) was being deleted and has been resolved.
2. List (mutable object) was being set as a default of a function argument.
3. If...then...else reduction.
4. Code formatting.

### CODE REVIEW

**Comment by sandipagarwal** [comment]
@sebdah Can we please get this rolled out. This is important and urgent. I don't know how to mark this a showstopper.

File: git-pylint-commit-hook [hunk 1]
```diff
@@ -13,7 +13,7 @@ LICENSE:
 """
 
 import argparse
-import os
+# import os
```
Comment (sebdah) [inline git-pylint-commit-hook]: Remove instead of commenting out.

File: git-pylint-commit-hook [hunk 2]
```diff
@@ -72,8 +72,8 @@ def main():
         sys.exit(0)
 
     # Remove the index.lock when it's exists
-    if os.path.exists('.git/index.lock'):
-        os.remove('.git/index.lock')
+    # if os.path.exists('.git/index.lock'):
+    #     os.remove('.git/index.lock')
```
Comment (sebdah) [inline git-pylint-commit-hook]: Remove instead of commenting out.

File: git-pylint-commit-hook [hunk 3]
```diff
@@ -72,8 +72,8 @@ def main():
         sys.exit(0)
 
     # Remove the index.lock when it's exists
```
Comment (sebdah) [inline git-pylint-commit-hook]: This comment should also be removed.

File: git_pylint_commit_hook/commit_hook.py [hunk 4]
```diff
@@ -188,23 +188,26 @@ def check_repo(
...
+        pylintrc = pylint_config.find_pylintrc() or '.pylintrc'
 
     # Stash any unstaged changes while we look at the tree
     with _stash_unstaged():
         # Find Python files
         for filename in _get_list_of_committed_files():
             try:
-                if _is_python_file(filename) and \
-                    not _is_ignored(filename, ignored_files):
+                if _is_python_file(filename) and not _is_ignored(filename, ignored_files):
```
Comment (sebdah) [inline git_pylint_commit_hook/commit_hook.py]: Why is this done? It seems like this line exceeds 80 chars. Please revert.

File: git-pylint-commit-hook (hunk 1 above)
Comment (sandipagarwal) [inline git-pylint-commit-hook]: Done.

File: git-pylint-commit-hook (hunk 3 above)
Comment (sandipagarwal) [inline git-pylint-commit-hook]: Done.

File: git-pylint-commit-hook (hunk 2 above)
Comment (sandipagarwal) [inline git-pylint-commit-hook]: Done.

File: git_pylint_commit_hook/commit_hook.py (hunk 4 above)
Comment (sandipagarwal) [inline git_pylint_commit_hook/commit_hook.py]: My bad. Editor was configured so.
Done.

**Comment by sandipagarwal** [comment]
Thanks @sebdah. I have fixed all review comments from you.

**Comment by sebdah** [comment]
This has now been released in version 2.2.1.

Thanks again @sandipagarwal for the submission!

**Comment by sandipagarwal** [comment]
Thanks @sebdah. :)

**Pull request merged**